src/GafferScene/EvaluateLightLinks.cpp

```diff
@@ -142,19 +146,21 @@ IECore::ConstCompoundObjectPtr EvaluateLightLinks::computeAttributes( const Scen
 	{
 		result->members()[ m_lightLinkAttrName ] = expressionToLightNames( illuminationExpressionData->readable(), inPlug(), lightsSet );
 	}
-	else
+	else if( path.size() == 1 )
 	{
-		// \todo: this transformation should probably get cached?
 		ConstPathMatcherDataPtr defaultLightsSetData = inPlug()->set( m_defaultLightSetName );
 		const PathMatcher &defaultLightsSet = defaultLightsSetData->readable();
 
-		IECore::StringVectorDataPtr defaultLightNames = new IECore::StringVectorData();
-		std::vector<std::string> &defaultLightNamesWritable = defaultLightNames->writable();
+		if( defaultLightsSet != lightsSet )
```

Comment:
This comparison is more expensive than the hash comparison used in `hashAttributes()`, so it's probably worth performing the cheap hash test first, and only doing the more expensive test if that fails.